spring-cloud-data-admin/src/main/java/org/springframework/cloud/data/admin/repository/RedisStreamDefinitionRepository.java

```diff
@@ -0,0 +1,196 @@
+		List<String> definitions = hashOperations.multiGet(names);
+		List<StreamDefinition> result = zipToStreamDefinitions(names, definitions);
+		long total = hashOperations.size();
+		return new PageImpl<>(result, pageable, total);
+	}
+
+	@Override
+	public <S extends StreamDefinition> S save(S entity) {
+		zSetOperations.add(entity.getName(), 0);
+		hashOperations.put(entity.getName(), entity.getDslText());
+		return entity;
+	}
+
+	@Override
+	public <S extends StreamDefinition> Iterable<S> save(Iterable<S> entities) {
+		Map<String, String> asMap = new HashMap<>();
+		Set<ZSetOperations.TypedTuple<String>> asScores = new HashSet<>();
+		for (StreamDefinition sd : entities) {
+			asMap.put(sd.getName(), sd.getDslText());
+			asScores.add(new DefaultTypedTuple<>(sd.getName(), 0D));
+		}
+		zSetOperations.add(asScores);
+		hashOperations.putAll(asMap);
+		return entities;
+	}
+
+	@Override
+	public StreamDefinition findOne(String s) {
+		String dsl = hashOperations.get(s);
+		return dsl != null ? new StreamDefinition(s, dsl) : null;
+	}
+
+	@Override
+	public boolean exists(String s) {
+		return hashOperations.hasKey(s);
+	}
+
+	@Override
+	public Iterable<StreamDefinition> findAll() {
+		return findAll(new Sort("name"));
+	}
+
+	@Override
+	public Iterable<StreamDefinition> findAll(Iterable<String> strings) {
+		Collection<String> keys;
+		if (strings instanceof Collection) {
+			keys = (Collection<String>) strings;
+		} else {
+			keys = new ArrayList<>();
+			for (String name : strings) {
+				keys.add(name);
+			}
+		}
+		List<String> dslTexts = hashOperations.multiGet(keys);
+		return zipToStreamDefinitions(keys, dslTexts);
+	}
+
+	@Override
+	public long count() {
+		return hashOperations.size();
+	}
+
+	@Override
+	public void delete(String s) {
+		hashOperations.delete(s);
+		zSetOperations.remove(s);
+	}
+
+	@Override
+	public void delete(StreamDefinition entity) {
+		delete(entity.getName());
+	}
+
+	@Override
+	public void delete(Iterable<? extends StreamDefinition> entities) {
+		Object[] keys;
+		if (entities instanceof Collection) {
+			keys = new Object[((Collection)entities).size()];
+			int i = 0;
+			for (StreamDefinition sd : entities) {
+				keys[i++] = sd.getName();
+			}
+		} else {
+			List<String> names = new ArrayList<>();
+			for (StreamDefinition sd : entities) {
+				names.add(sd.getName());
+			}
+			keys = names.toArray();
+		}
+		hashOperations.delete(keys);
+		zSetOperations.remove(keys);
+	}
+
+	@Override
+	public void deleteAll() {
+		hashOperations.getOperations().delete(hashKey);
+		zSetOperations.getOperations().delete(hashKey + ".names");
+	}
+
+	/**
+	 * Return a list of StreamDefinitions, made by mapping a set of non null names and possibly null dsl texts.
+	 * In case of null dsl text, a null element is added to the result list.
+	 */
+	private List<StreamDefinition> zipToStreamDefinitions(Collection<String> names, Collection<String> definitions) {
```

Comment:
this signature could change to (List, List) since the caller can pass a List for both 'keys' and 'dslTexts' (just remove the `if (entities instanceof Collection)` block or change Collection to List for that too).